featurex/converters/video.py

```diff
@@ -0,0 +1,79 @@
+from featurex.stimuli.video import VideoStim, DerivedVideoStim, VideoFrameStim
+from featurex.converters import Converter
+
+import pandas as pd
+
+
+class VideoToDerivedVideoConverter(Converter):
+
+    ''' Base AudioToText Converter class; all subclasses can only be applied to
+    audio and convert to text. '''
+    target = VideoStim
+    _output_type = DerivedVideoStim
+
+
+class FrameSamplingConverter(VideoToDerivedVideoConverter):
+    ''' 
+    Samples frames from video stimuli, to improve efficiency
+
+    Args:
+        every (int): takes every nth frame
+        hertz (int): takes n frames per second
+        num_frames (int): takes top n frames sorted by the absolute difference
+         with the next frame
+    '''
+    def __init__(self, every=None, hertz=None, num_frames=None):
+        self.every = every
+        self.hertz = hertz
+        self.num_frames = num_frames
+
+    def _convert(self, video):
+        if not hasattr(video, "frame_index"):
+            frame_index = range(video.n_frames)
+        else:
+            frame_index = video.frame_index
+        
+        if not hasattr(video, "history"):
+            history = pd.DataFrame(columns=["filter", "value", "n_frames"])
+        else:
+            history = video.history
+
+        if self.every is not None:
+            new_idx = range(video.n_frames)[::self.every]
+            history.loc[history.shape[0]]= ["every", self.every, len(new_idx)]
```

Comment:
What I mean is that it's not clear why we check the `VideoStim` to see if it already has a `history` attribute (line 36). This would make sense if we want to essentially chain histories, but unless I'm mistaken, it's going to cause problems in cases where there already *is* a history, and we're now updating only some of the rows (per the `.loc` call, which will leave rows with non-overlapping existing index values untouched). My suggestion is to basically wipe the history each time (i.e., not to check for the video's existing history), so that all rows in the `.history` attribute only relate to the very last conversion. Does that make sense?